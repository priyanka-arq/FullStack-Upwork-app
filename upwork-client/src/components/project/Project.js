import React, { Component } from "react";
import { connect } from "react-redux";
import ProjectList from "./ProjectList";

export class Project extends Component {
  render() {
    return (
      <div>
        <ProjectList projects={this.props.projects} />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    projects: state.projects,
  };
};

export default connect(mapStateToProps)(Project);
