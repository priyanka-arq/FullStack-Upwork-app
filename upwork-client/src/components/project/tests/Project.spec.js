import React from "react";
import { Project } from "../Project";
import { shallow } from "enzyme";
import projects from "../../../fixtures/projects";
let wrapped;

beforeEach(() => {
  wrapped = shallow(<Project projects={projects} />);
});

afterEach(() => {
  wrapped.unmount();
});

test("Should create snapshot", () => {
  expect(wrapped).toMatchSnapshot();
});

test("Should render ProjectList component", () => {
  expect(wrapped.find("ProjectList").length).toEqual(1);
});
