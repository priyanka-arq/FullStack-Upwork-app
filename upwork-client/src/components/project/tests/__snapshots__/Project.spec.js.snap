// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create snapshot 1`] = `
<div>
  <ProjectList
    projects={
      Array [
        Object {
          "_id": 1,
          "currentUser": Object {
            "id": 1,
          },
          "description": "TEST1",
          "email": "TEST@ga.com",
          "estimatedBudget": 454,
          "name": "TEST1",
          "paymentType": "Fixed-Price",
          "phone": "056565566",
          "skills": Array [
            "React",
            "Node",
          ],
        },
        Object {
          "_id": 2,
          "description": "TEST2",
          "email": "TEST@ga.com",
          "estimatedBudget": 454,
          "name": "TEST2",
          "paymentType": "Fixed-Price",
          "phone": "056565566",
          "skills": Array [
            "Java",
            "CSS",
          ],
          "user": 2,
        },
        Object {
          "_id": 3,
          "description": "TEST1",
          "email": "TEST@ga.com",
          "estimatedBudget": 454,
          "name": "TEST3",
          "paymentType": "Fixed-Price",
          "phone": "056565566",
          "skills": Array [
            "HTML",
            "Angular",
          ],
          "user": 3,
        },
      ]
    }
  />
</div>
`;
